vite.config.js

```diff
@@ -0,0 +1,9 @@
+import { resolve } from 'node:path'
+import viteReact from '@vitejs/plugin-react'
+
+/** @type {import('vite').UserConfig} */
+export default {
+  base: '/',
+  root: resolve(import.meta.dirname, 'src/client'),
+  plugins: [viteReact()]
```

Comment:
Since you're using `tsc` to build the server code into the `dist` directory, I personally think it's kind of weird to have the client code built to `src/client/dist`. I prefer to add the following here to the vite config:

```
build: {
  emptyOutDir: true, // optional
  outDir: resolve(import.meta.dirname, 'dist/client'),
}
```

Additionally, I'm trying to get [this PR](https://github.com/fastify/fastify-vite/pull/156) into `fastify-vite` which gives another vite plugin called `viteFastify()` that helps with production builds.